test/components/componentTests.ts

```diff
@@ -1,74 +1,87 @@
 ///<reference path="../testReference.ts" />
 
-function assertComponentXY(component: Plottable.Component, x: number, y: number, message: string) {
-  // use <any> to examine the private variables
-  let translate = d3.transform((<any> component)._element.attr("transform")).translate;
-  let xActual = translate[0];
-  let yActual = translate[1];
-  assert.strictEqual(xActual, x, "X: " + message);
-  assert.strictEqual(yActual, y, "Y: " + message);
-}
-
-describe("Component behavior", () => {
-  let svg: d3.Selection<void>;
+describe("Component", () => {
   let c: Plottable.Component;
   let SVG_WIDTH = 400;
   let SVG_HEIGHT = 300;
+  let svg: d3.Selection<void>;
   beforeEach(() => {
-    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
     c = new Plottable.Component();
+    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
```

Comment:
No nested beforeEach() please. Can't find my arguments in the issue linked by @ztsai . However the point is that having nested beforeEach, we are losing track of variables, and tests get so much harder to read.